### ISSUE TITLE
Offer a way to configure tabs?

### ISSUE THREAD

**Issue by 3oheicrw** [snaildos/Fifo-Browser#112]
For example, I want tabs to have behaviors like on Chrome/Firefox. For example, when you click a bookmark in bookmark bar, it will open in the current tab instead of in a new tab. Another example: when you open a link with middle mouse button, it will open in a new tab but don't switch to this tab immediately like it currently does. Also, when you right click on a link, there are currently two duplicate "Open in new tab", the first open the link in new tab and switch to it immediately, the second also open the link in new tab but don't switch to it immediately like the first. Since both of them have the same name, it will cause confusion for some people. I suggest changing it to something more easier to understand. But the better solution IMO is allow the users to configure the behaviors of tabs, so you don't have to include two "Open in new tab" that serves different way of open it new tab like that. Only one "Open in new tab" should be provided, and the users will decide whether to switch to the new tab immediately or not.

Currently I have these three options for a tab in mind: open in current tab, open in foreground tab (new tab) and open in background tab (new tab). Let the users to set there desired behaviors. e.g: when open bookmarks, will be choose from: open in current tab, open in foreground tab (new tab switch to immediately), open in background tab (new tab but don't switch to immediately).

**Comment by 3oheicrw** [comment]
"Close tabs to the right" also has a duplicated entry. It seems they served the same purpose as the two "Open in new tab". Tabs opened with the first "Open in new tab" will be closed with the first "Close tabs to the right" (I guess, not tested). Tabs opened with the second "Open in new tab" will be closed with the second "Close tabs to the right" (tested, know for sure it works, it also close tab manually opened by me but not by "Open in new tab"). 

Ironical enough, "Close tabs to the right" is provided but there is no "Close tabs to the left" (equally useful) and "Close other tabs" (the most useful). I think you should copy the well established behaviors of Chrome/Firefox my friend. Doing everything from scratch your way only caused you more problems.

**Comment by snaildos** [comment]
> "Close tabs to the right" also has a duplicated entry. It seems they served the same purpose as the two "Open in new tab". Tabs opened with the first "Open in new tab" will be closed with the first "Close tabs to the right" (I guess, not tested). Tabs opened with the second "Open in new tab" will be closed with the second "Close tabs to the right" (tested, know for sure it works, it also close tab manually opened by me but not by "Open in new tab").
> 
> Ironical enough, "Close tabs to the right" is provided but there is no "Close tabs to the left" (equally useful) and "Close other tabs" (the most useful). I think you should copy the well established behaviors of Chrome/Firefox my friend. Doing everything from scratch your way only caused you more problems.

Sadly the tab system is by scratch, however I will be copying their context menu for the tabs. 
I do thank this feedback so much.
Thank you!

**Comment by 3oheicrw** [comment]
IMO there is nothing wrong for us to copy the well established behaviors of popular browsers like Chrome and Firefox. We cherry pick the best from each of them but don't blindly copy them, if doing so we are pretty much become a clone of them :smile:

**Comment by 3oheicrw** [comment]
BTW, I think you should consider copy Firefox's way to dealing with too many tabs. I think it's better than Chrome. Firefox will not try to fit them all in the visible space of the tab bar like Chrome but let them expand indefinitely (Fifo currently also does so), but the Firefox people also provide an arrow button for we to scroll through the many tabs (just try with Firefox and get your own description, my English sucks I know most of the time it can't describe what I'm really want to tell you). Fifo currently lacks this arrow button. This make scrolling through the many tabs become very difficult.

**Comment by snaildos** [comment]
Yeah. I am going to. The context menu for tabs and sites are very wonky at this current stage. Thank you so much for the feedback. Our next major release will be the Customization update.

**Comment by DI555** [comment]
@snaildos , could you please, make a close icon on tabs be optional ?
i mean smth. with options like "no_Xicons_on_all_tabs", "Xicon_on_active_tab", "Xicons_on_inactive_tabs" and "Xicons_on_all_tabs"..

just..i mean.. many of us don't use close icons on tabs at all ((..
and would be great to see such things optionable, please!

**Comment by snaildos** [comment]
> @snaildos , could you please, make a close icons on tabs be optional ? i mean smth. with options like "no_Xicons_on_all_tabs", "Xicon_on_active_tab", "Xicons_on_inactive_tabs" and "Xicons_on_all_tabs"..
> 
> just..i mean.. many of us don't use close icons on tabs at all ((.. and would be great to see such things optionable, please!

Yeah, that wont be happening. That is a ton of work.

**Comment by DI555** [comment]
@snaildos , i see :(
or may be, could you please, maybe just one option of that, -
"it's turn off all closebutton icons" ,
 - for those who using midle mouse button to closing tabs, and afraid of click to "X" somehow..
 -

**Comment by va0dl914** [comment]
> @snaildos , i see :( or may be, could you please, maybe just one option of these, - "it's turn off all closebutton icons" ,
> 
> * for those who using midle mouse button to closing tabs, and afraid of click to "X" somehow at work with tabs...

Nonsense. I think you are using Ctrl + W like me until I found middle mouse. You will use the mouse anyway.

**Comment by snaildos** [comment]
I'll look into it.
